composer.json

```diff
@@ -27,7 +27,7 @@
         "doctrine/cache": "^1.12.1 || ^2.1.1",
         "doctrine/collections": "^1.5",
         "doctrine/common": "^3.0.3",
-        "doctrine/dbal": "^3.2",
+        "doctrine/dbal": "^3.3@dev",
```

Comment:
I need the abstract middleware classes for now. We can drop the `@dev` as soon as DBAL 3.3 is released.

Reply:
This is not `require-dev` though, are you sure you want to do this?

Reply:
I believe it's fine to keep it this way before the release.

Reply:
All static analysis issues seem related to the dependency update (no longer existing baseline errors in PHPStan and the deprecation error in Psalm).

Reply:
> This is not `require-dev` though, are you sure you want to do this?

I believe that DBAL 3.3 will be released way before ORM 3.0.0, so I think it's okay to do this temporarily.

Reply:
Oh right this is ORM 3, looks totally fine indeed!